src/lib/y2firewall/importer_strategies/suse_firewall.rb

```diff
@@ -0,0 +1,214 @@
+
+require "y2firewall/firewalld"
+
+module Y2Firewall
+  module ImporterStrategies
+    # This class is reponsible of parsing SuSEFirewall2 firewalld profile's
+    # section configuring the Y2Firewall::Firewalld instance according to it.
+    class SuseFirewall
+      # [Hash] AutoYaST profile firewall's section
+      attr_accessor :profile
+
+      # SuSEFirewall2 zones
+      ZONES = ["DMZ", "INT", "EXT"].freeze
+
+      # @return [Array<string>] list of zones
+      def zones
+        ZONES
+      end
+
+      # Constructor
+      #
+      # @param [Hash] AutoYaST profile firewall's section
+      def initialize(profile)
+        @profile = profile
+      end
+
+      # It process the profile configuring the firewalld zones that match
+      # better with the SuSEFirewalld ones.
+      def import
+        return true if profile.empty?
+
+        zones.each { |z| process_zone(z) }
+
+        if ipsec_trust_zone
+          zone = firewalld.find_zone(zone.equivalent(ipsec_trust_zone))
+          zone.services << "ipsec"
+        end
+
+        firewalld.log_denied_packets = log_denied_packets
+
+        true
+      end
+
+      # Given a SuSEFirewall2 zone name it process the profile's configuration
+      # corresponding to that zone configuring the equivalent firewalld zone
+      # object.
+      # @param name [String] SuSEFirewall2 zone name
+      def process_zone(name)
+        zone = firewalld.find_zone(zone_equivalent(name))
+
+        zone.interfaces = interfaces(name) if interfaces(name)
+        zone.services   = services(name)   if services(name)
+        zone.ports      = ports(name)      if ports(name)
+      end
+
+      # Obtain the services for the given SuSEFIrewall2 zone name from the
+      # profile.
+      #
+      # @param zone_name [String]
+      def services(zone_name)
+        services = profile["FW_CONFIGURATIONS_#{zone_name}"]
+
+        services ? services.split(" ") : nil
+      end
+
+      # Obtain the interfaces for the given SuSEFIrewall2 zone name from the
+      # profile.
+      #
+      # @param zone_name [String]
+      def interfaces(zone_name)
+        interfaces = profile["FW_DEV_#{zone_name}"]
+
+        interfaces ? interfaces.split(" ") : nil
+      end
+
+      # Obtain the ports for the given SuSEFIrewall2 zone name from the
+      # profile.
+      #
+      # @param zone_name [String]
+      def ports(zone)
+        return nil unless ip_ports(zone) || rpc_ports(zone) || tcp_ports(zone) || udp_ports(zone)
+
+        [ip_ports(zone), rpc_ports(zone), tcp_ports(zone), udp_ports(zone)].compact.flatten
+      end
+
+    private
+
+      # Obtain the IP ports for the given SuSEFIrewall2 zone name from the
+      # profile.
+      #
+      # @param zone_name [String]
+      # @return [Array<Strint>, nil] list of configured IPP ports; nil if no
+      # configured
+      def ip_ports(zone)
+        ports = profile["FW_SERVICES_#{zone}_IP"]
```

Comment:
Maybe you would save this value in an instance variable. It is not a big deal, but you are calling this method twice from `#ports` and I guess that you do not expect the profile to change.